features/docs.feature

```diff
@@ -0,0 +1,189 @@
+@cli @pc-render
+Feature: `pc render` command
+
+  Background: Sandbox
+    Given I am in "/tmp/sandbox/behave" directory
+    Given I have temporary $HOME in "/tmp/sandbox/home"
+
+  @docs-tutorial
+  Scenario: Tutorial
+    When I run "pc init -p"
+    Then the command should exit with a status code of "0"
+    When I run command:
+      """
+      cat <<EOF > partcad.yaml
+      import:
+        # Public PartCAD repository (reference it explicitly if required)
+        pub:
+          type: git
+          url: https://github.com/partcad/partcad-index.git
+      EOF
+      """
+    Then the command should exit with a status code of "0"
+    When I run "pc list"
+    Then the command should exit with a status code of "0"
+    When I run "echo "translate (v= [0,0,0])  cube (size = 10);" > test.scad"
+    Then the command should exit with a status code of "0"
+    When I run "pc add part scad test.scad"
+    Then the command should exit with a status code of "0"
+    When I run "pc inspect :test"
+    Then the command should exit with a status code of "0"
+    When I run "pc render -t stl :test"
+    Then the command should exit with a status code of "0"
+
+
+  @docs-design
+  Scenario: Design: Parametrized objects
+    When I run "pc init"
+    Then the command should exit with a status code of "0"
+    When I run command:
+      """
+      pc \
+        inspect \
+          -p length=30 \
+          -p size=M4-0.7 \
+          /pub/std/metric/cqwarehouse:fastener/hexhead-din931
+      """
+    # TODO: Failure to pip install from GH repo
+    Then the command should exit with a status code of "1"
+    When I run command:
+      """
+      pc \
+        inspect \
+          /pub/std/metric/cqwarehouse:fastener/hexhead-din931;length=30,size=M4-0.7
+      """
+    Then the command should exit with a status code of "0"
+     And STDOUT should contain 'DONE: InitCtx:'
+     And STDOUT should contain 'Visualizing in "OCP CAD Viewer"...'
+     And STDOUT should contain 'Unable to open config file /tmp/sandbox/home/partcad-cli-'
+     And STDOUT should contain 'No VS Code or "OCP CAD Viewer" extension detected.'
+     And STDOUT should contain 'DONE: inspect: this'
+
+   @docs-design
+  Scenario: Design: Objects in a cart
+    When I run "pc init"
+    Then the command should exit with a status code of "0"
+    When I run command:
+      """
+      pc supply quote \
+        --provider /pub/svc/commerce/gobilda:gobilda \
+        /pub/robotics/multimodal/openvmp/robots/don1:assembly-wormgear#10
+      """
+    # TODO: We don't have enough XXX stock on hand for the quantity you selected
+    Then the command should exit with a status code of "1"
```

Comment:
Well... now I start thinking that during tests we don't really need to go to the live provider and can get avail with simple local HTTP mock (could be started as python script). This will make those tests reproducible, fast and wont lead to billing.